### PR TITLE
chore(RHTAPWATCH-1110): solve KubeLinter's resource allocation issues

### PR DESCRIPTION
**Pull request by ShakedAviv50313**
Allocating resources to solve Kubelinter's resource allocation issues.

I tested those changed during and after all changes were made, I was also able to release using my local Konflux-ci.

**Resource allocation tests and results:**
**dependencies/cert-manager/Kustomization.yml + dependencies/keycloak/deployment/db.yaml:**

Raised all cpu and memory requests from 10m to 90m and from 10Mi to 90Mi accordingly.

Tests tried : 30m,30Mi - build-container step failed from insufficient resources, same happened for 50. for 80 and 85 build-container passed but the next steps were flaky and each time different step failed because of lacking resources.

**dependencies/keycloak/deployment/kustomization.yml:**
cpu and memory set to 50

**dependencies/pipelines-as-code/kustomization.yml:**
cpu and memory set to 50

**Testing release with all the values above - success**

**dependencies/pre-deployment-pvc-binding/pre-deployment-pvc-binding.yaml:**
requests:
cpu: 10m
memory: 50Mi
limits:
cpu: 50m
memory: 100Mi

**dependencies/registry/registry.yaml:**
requests:
cpu: 10m
memory: 50Mi
limits:
cpu: 100m
memory: 250Mi

**dependencies/tekton-operator/kustomization.yml:**
requests:
cpu: 10m
memory: 10Mi
limits:
cpu: 100m
memory: 100Mi
 
**Testing release with all the values above - success**

**dependencies/tekton-results/kustomization.yml:**
requests:
cpu: 10m
memory: 10Mi
limits:
cpu: 100m
memory: 100Mi

**Testing release with all the values above - success**

### CODE REVIEW

File: dependencies/keycloak/deployment/kustomization.yml [hunk 1]
```diff
@@ -23,3 +23,16 @@ patches:
...
+  - patch: |
+      - op: add
+        path: /spec/template/spec/containers/0/resources
+        value:
+          requests:
+            cpu: 50m
+            memory: 50Mi
+          limits:
+            cpu: 100m
+            memory: 100Mi
```
Comment (yftacherzog) [inline dependencies/keycloak/deployment/kustomization.yml]: @gbenhaim do you expect this to be enough?

Comment (ShakedAviv50313) [reply dependencies/keycloak/deployment/kustomization.yml]: i see now that the sanity test failed with the following error:
```kubectl wait --for=condition=Ready --timeout=300s -n keycloak -l app.kubernetes.io/name=keycloak-operator pod: Keycloack did not become available within the allocated time.```

I'll increase the amount to see if it solves the issue (my guess is that keycloak didn't have enough memory or cpu to become available in time)

Comment (yftacherzog) [reply dependencies/keycloak/deployment/kustomization.yml]: Yes. Keycloak is quite heavy.

**Comment by yftacherzog** [comment]
@ShakedAviv50313 what's the output of `kubectl describe nodes`? Specifically, the `Allocated resources` part at the bottom?

**Comment by ShakedAviv50313** [comment]
on m computer right now:

Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests      Limits
  --------           --------      ------
  cpu                2120m (17%)   11 (91%)
  memory             4474Mi (22%)  15198Mi (77%)
  ephemeral-storage  0 (0%)        0 (0%)
  hugepages-1Gi      0 (0%)        0 (0%)
  hugepages-2Mi      0 (0%)        0 (0%)
Events:              <none>

**Comment by yftacherzog** [comment]
> @yftacherzog on my computer right now:
> 
> Allocated resources: (Total limits may be over 100 percent, i.e., overcommitted.) Resource Requests Limits
> 
> cpu 2120m (17%) 11 (91%) memory 4474Mi (22%) 15198Mi (77%) ephemeral-storage 0 (0%) 0 (0%) hugepages-1Gi 0 (0%) 0 (0%) hugepages-2Mi 0 (0%) 0 (0%) Events:

thanks

File: dependencies/keycloak/deployment/kustomization.yml (hunk 1 above)
Comment (ShakedAviv50313) [reply dependencies/keycloak/deployment/kustomization.yml]: I raised the quota to:
```         
 requests:
            cpu: 200m
            memory: 200Mi
          limits:
            cpu: 300m
            memory: 300Mi
```

Comment (ShakedAviv50313) [reply dependencies/keycloak/deployment/kustomization.yml]: and now it works, the sanity test succeeded

**Comment by yftacherzog** [comment]
@gbenhaim @psturc can you have a look to check if you see any potential issues with the suggested resource quotas?

**Review by yftacherzog — APPROVED** [review]
lgtm

**Pull request merged**